proxy/src/main.rs

```diff
@@ -126,6 +141,19 @@ async fn main() -> anyhow::Result<()> {
         )));
     }
 
+    if let Some(metric_collection_endpoint) = &config.metric_collection_endpoint {
+        let hostname = hostname::get().unwrap().into_string().unwrap();
```

Comment:
Please don't use unwrap. This function returns anyhow::Result. Either use `.context(...)?` or just `?`.